packages/x-date-pickers-pro/src/internal/hooks/useMultiInputRangeField/useMultiInputDateTimeRangeField.ts

```diff
@@ -59,24 +65,36 @@ export const useMultiInputDateTimeRangeField = <TDate, TChildProps extends {}>({
   TChildProps
 >): UseMultiInputRangeFieldResponse<TChildProps> => {
   const sharedProps = useDefaultizedDateTimeRangeFieldProps<TDate, TChildProps>(inSharedProps);
+  const adapter = useLocalizationContext<TDate>();
 
   const { value: valueProp, defaultValue, format, onChange } = sharedProps;
 
   const firstDefaultValue = React.useRef(defaultValue);
 
   // TODO: Maybe export utility from `useField` instead of copy/pasting the logic
-  const buildChangeHandler = (index: 0 | 1) => {
+  const buildChangeHandler = (
```

Comment:
That part could be centralized between the 3 multi input fields.
It would be a first part of shared logic.

Reply:
I agree, it look duplicated, but this function use its context. If you extract it, you will have to pass `firstDefaultValue`, `dateRangePickerValueManager`, ... as parameters and it will be harder to read

Reply:
True, maybe we can try to refacto later to see if we can reach a better trade-off
But starting with a duplication avoid to build bad abstractions